flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/publisher/fanout/StreamConsumerRegistrar.java

```diff
@@ -77,8 +79,8 @@ public StreamConsumerRegistrar(
      * @throws InterruptedException
      */
     public String registerStreamConsumer(final String stream, final String streamConsumerName)
-            throws ExecutionException, InterruptedException {
-        LOG.debug("Registering stream consumer - {}::{}", stream, streamConsumerName);
+            throws Exception {
```

Comment:
Why did you make this change? This will possible change the error handling behaviour

Reply:
The main reason of the change is that we are using SyncClient for registerStreamConsumer, so it is not a FutureTask and it won't throw ExecutionException. However, I agree that InterruptedException should be readded as it might be thrown during registrationBackoff

Reply:
I agree with @dannycranmer , we would better not generalize the Exception.

Reply:
I think we can generalise it here because previously ExecutionException, InterruptedException were thrown from a CompletableFuture get method. However, we are no longer using Future here as registerStreamConsumer is called synchronously now (via Callable call method) instead of async. Unlike CompletableFuture, Callable call method throws Exception instead of Execution + InterruptedException.

Reply:
My concerns is the Exception is a wider catch bucket than before. However given `ExecutionException, InterruptedException` would have caught the majority and the parent wraps in a `FlinkKinesisStreamConsumerRegistrarException`, it looks ok